Bio/GenBank/Scanner.py

```diff
@@ -1322,7 +1322,19 @@ def _feed_first_line(self, consumer, line):
             else:
                 consumer.residue_type(line[44:63].strip())
 
-            consumer.molecule_type(line[44:54].strip())
+            molecule_type = line[44:54].strip()
+
+            # Writing out records will fail if we have a lower case DNA
+            # or RNA string in here, so upper case it.
+            # This is a bit ugly, but we don't want to upper case e.g.
+            # the m in mRNA, but thanks to the strip we lost the spaces
+            # so we need to index from the back
+            if molecule_type and not molecule_type[-3:].isupper():
+                warnings.warn("Non-upper case molecule type in LOCUS line: %s"
+                              % molecule_type, BiopythonParserWarning)
+                molecule_type = "".join(molecule_type[:-3] + molecule_type[-3:].upper())
```

Comment:
The minimally invasive fix would be to change the assert in the writer to match the assert in the parser. That would mean we also write an invalid lower-case `dna` string.

Reply:
After all in https://github.com/biopython/biopython/blob/master/Bio/GenBank/Scanner.py#L1284 ff. we say:
```python
assert line[47:54].strip() == "" \
                or 'DNA' in line[47:54].strip().upper() \
                or 'RNA' in line[47:54].strip().upper(), \
'LOCUS line does not contain valid sequence type (DNA, RNA, ...):\n' + line
```
but in the writer https://github.com/biopython/biopython/blob/master/Bio/SeqIO/InsdcIO.py#L686 we have:
```python

        assert line[47:54].strip() == "" \
            or 'DNA' in line[47:54].strip() \
            or 'RNA' in line[47:54].strip(), \
'LOCUS line does not contain valid sequence type (DNA, RNA, ...):\n' + line
```

Reply:
I don't have strong opinions on how to fix this, as long as I can get a version that will not fall over on writing a file that I analysed for some hours because of something that didn't even trigger a warning when parsing it.

Reply:
Let's (1) add the new parser warning, and (2) downgrade that writer assert to warning.

Optionally (3) automatically convert to upper case on output with the warning.